addons/sourcemod/scripting/gokz-replays/playback.sp

```diff
@@ -937,12 +938,15 @@ void PlaybackVersion2(int client, int bot, int &buttons)
 			EmitSoundToClientSpectators(client, gC_ModeEndSounds[GOKZ_GetCoreOption(client, Option_Mode)]);
 		}
 
-		// Set velocity to travel from current origin to recorded origin
-		float currentOrigin[3], velocity[3];
-		Movement_GetOrigin(client, currentOrigin);
-		MakeVectorFromPoints(currentOrigin, currentTickData.origin, velocity);
-		ScaleVector(velocity, 1.0 / GetTickInterval());
-		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, velocity);
+		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, currentTickData.velocity);
+		// TeleportEntity does not set the absolute origin and velocity so we need to do it
+		// to prevent inaccurate eye position interpolation.
+		SetEntPropVector(client, Prop_Data, "m_vecAbsVelocity", currentTickData.velocity);
+		
+		SetEntPropVector(client, Prop_Data, "m_vecAbsOrigin", currentTickData.origin);
```

Comment:
Setting the origin directly instead of changing only the velocity was usually the cause of stuttering in the past, would it be enough to fix the issue if we only set the abs velocities and eye angles?

Reply:
The current stuttering in 3.3.0 is caused by client interpolation not working on fake clients that aren't bots. GameChaos's SimpleTAS also does a similar thing so I believe it should be sufficient.

For the record, the [kz_kiwiterror run on youtube](https://www.youtube.com/watch?v=fa8fZ8J56hg) is recorded with the code in this PR. Of course you should also test it for yourself as well, as I might miss some other edge cases.

Reply:
I don't know how `m_vecAbsOrigin` really behaves, I just know that setting the regular origin every tick is no good, so it raised a red flag. If it is required to fix the stutters we'll leave it, I'll do some testing as well.

Reply:
I think as long as the game doesn't add the "no interpolation" flag to the bot (might be the case with `TeleportEntity` using the `origin` argument), it should be fine, hopefully.

Reply:
damn you should not set their origin by every ticks, or your bot will be choppy whatever it's fakeclient or not.
here's a simple solution: https://github.com/shavitush/bhoptimer/blob/master/addons/sourcemod/scripting/shavit-replay-playback.sp#L2677-L2689

Reply:
I doubt that changing one netprop would cause a lot of lag, and it's definitely less expensive than TeleportEntity with an origin argument (and it only changes the local origin which is insufficient as well, as described in the PR)

I could in theory check for the predicted origin vs the actual replay origin and teleport them when necessary but then the extra comparisons would cost more resources either way.

Reply:
I noticed that the bot is a bit choppy after setting origin(netprop) on every tick. Haven't you noticed?

Reply:
The latest commit should make the bot much less choppy now.